users/serializers.py

```diff
@@ -257,4 +255,60 @@ def create(self, validated_data):
         """
         self.request.user.set_password(validated_data.get("new_password"))
         self.request.user.save()
-        return self.request.user
\ No newline at end of file
+        return self.request.user
+
+
+class TimeLogSerializer(serializers.Serializer):
+    """ timelog serializer for clock-in and clock-out
+        from slack api
+    """
+
+    user = None
+    team_id = serializers.CharField()
+    channel_id = serializers.CharField()
+    channel_name = serializers.CharField()
+    user_id = serializers.CharField()
+    
+    def validate_team_id(self, team_id):
+        """ validate if the team source of the
+            request came from the right team. (swiftkind)
+            if not, deny access.
+        """
+        if team_id != settings.SLACK_TEAM_ID:
+            raise serializers.ValidationError(
+                _('[Invalid Request] Wrong workspace.'),
+                code="invalid_request",
+            )
+
+        return team_id
+
+    def validate_user_id(self, user_id):
+        """ validate if the user is a valid member
+            of the workspace.
+        """
+        self.user = User.objects.filter(slack_id=user_id).first()
+        if not self.user:
+            raise serializers.ValidationError(
+                _('[Invalid Request] Requestor is not a registered user.'),
+                code="invalid_request",
+            )
+        
+        return user_id
+    
+    def create(self, validated_data):
+        """ create or update timelog instance
+        """
+        # get all timelog where user = request.user
+        # and clock out = Null
+        time_log = TimeLog.objects.filter(user=self.user, time_out=None)        
+        if time_log:
+            # update instances and add clock out  
+            for t_log in time_log:
+                TimeLog.objects.filter(id=t_log.id).update(time_out=timezone.now())
```

Comment:
why do you need to re-query the same instance?

this doesn't work?
`TimeLog.objects.filter(user=self.user, time_out=None).update(time_out=timezone.now())`

also do not return a string message. this function also doesn't make sense for me. i mean i don't understand the purpose of this function